src/zhinst/toolkit/control/node_tree.py

```diff
@@ -131,19 +154,97 @@ def _setter(self, value):
         if "Write" in self._properties:
             if value != self._cached_value:
                 if self._map is not None and isinstance(value, str):
-                    if value not in self._map.values():
+                    # Construct a list from all allowed values in the mapping
+                    allowed_values = self._flatten_mapping_values(self._map)
+                    if value not in allowed_values:
                         raise ToolkitNodeTreeError(
-                            f"The value '{value}' is not in {list(self._map.values())}."
+                            f"The value '{value}' is not in {allowed_values}."
                         )
-                    inv_map = {v: k for k, v in self._map.items()}
+                    inv_map = self._invert_mapping(self._map)
                     value = inv_map[value]
-                value = self._set_parser(value)
+                elif self._map is not None and isinstance(value, int):
+                    allowed_values = list(self._map.keys())
+                    if value not in allowed_values:
+                        raise ToolkitNodeTreeError(
+                            f"The value '{value}' is not in {allowed_values}."
+                        )
+                # If the set_parser is a list of callables, call them
+                # one by one inside a loop
+                if isinstance(self._set_parser, list):
+                    for callable_element in self._set_parser:
+                        value = callable_element(value)
+                else:
+                    value = self._set_parser(value)
                 self._device._set(self._path, value)
+                # After setting the value, read it back from the device
+                value = self._device._get(self._path)
```

Comment:
create 'check' argument for setter (default = True)

return (Bool, node_value)